spirv_glsl.cpp

```diff
@@ -6467,7 +6467,10 @@ void CompilerGLSL::emit_unary_func_op(uint32_t result_type, uint32_t result_id,
 void CompilerGLSL::emit_binary_func_op(uint32_t result_type, uint32_t result_id, uint32_t op0, uint32_t op1,
                                        const char *op)
 {
-	bool forward = should_forward(op0) && should_forward(op1);
+	// Opaque types (e.g. OpTypeSampledImage) must always be forwarded in GLSL unless they are a uniform variable
+	const auto &type = get_type(result_type);
+	bool must_forward = type_is_opaque_value(type) && type.storage != spv::StorageClassUniform;
```

Comment:
This comment is a bit misleading because StorageClassUniform (UBO / SSBO in SPIR-V 1.0) != StorageClassUniformConstant (descriptors). I think opaque types always have to be forwarded no matter what, just checking if type is opaque should be enough?

Reply:
I think so, I wasn't sure if the uniform storage class is handled differently here.